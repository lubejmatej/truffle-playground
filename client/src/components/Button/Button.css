.Button {
  border-radius: 0;
  outline: none;
  cursor: pointer;
  display: inline-block;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  text-decoration: none;
  vertical-align: middle;
  user-select: none;
  background-color: transparent;
  padding: 0.5rem 0.75rem;
  font-size: 1rem;
  box-shadow: none;
  border: 0;
}

.Button:not(:disabled) {
  cursor: pointer;
}

.Button--primary {
  color: var(--primary);
}

.Button--secondary {
  color: var(--warning);
}

.Button--primary:hover {
  color: var(--light);
}

.Button--secondary:hover {
  color: var(--light);
}
