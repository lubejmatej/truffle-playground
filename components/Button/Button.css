.Button {
  --button-color: var(--dark);

  border-radius: 0;
  outline: none;
  cursor: pointer;
  display: inline-block;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  text-decoration: none;
  vertical-align: middle;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  background-color: transparent;
  padding: 0.5rem 0.75rem;
  font-size: 1rem;
  box-shadow: none;
  border: 0;

  color: var(--button-color);
}

.Button:not(:disabled) {
  cursor: pointer;
}

.Button--outline {
  border: 1px solid currentColor;
}

.Button--primary {
  --button-color: var(--primary);
}

.Button--secondary {
  --button-color: var(--warning);
}

.Button--primary:hover,
.Button--secondary:hover {
  --button-color: var(--light);
}
