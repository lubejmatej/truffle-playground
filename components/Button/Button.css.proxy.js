// [snowpack] add styles to the page (skip if no document exists)
if (typeof document !== 'undefined') {
  const code = ".Button {\n  --button-color: var(--dark);\n\n  border-radius: 0;\n  outline: none;\n  cursor: pointer;\n  display: inline-block;\n  font-weight: 400;\n  line-height: 1.5;\n  text-align: center;\n  text-decoration: none;\n  vertical-align: middle;\n  -webkit-user-select: none;\n     -moz-user-select: none;\n      -ms-user-select: none;\n          user-select: none;\n  background-color: transparent;\n  padding: 0.5rem 0.75rem;\n  font-size: 1rem;\n  box-shadow: none;\n  border: 0;\n\n  color: var(--button-color);\n}\n\n.Button:not(:disabled) {\n  cursor: pointer;\n}\n\n.Button--outline {\n  border: 1px solid currentColor;\n}\n\n.Button--primary {\n  --button-color: var(--primary);\n}\n\n.Button--secondary {\n  --button-color: var(--warning);\n}\n\n.Button--primary:hover,\n.Button--secondary:hover {\n  --button-color: var(--light);\n}\n";

  const styleEl = document.createElement("style");
  const codeEl = document.createTextNode(code);
  styleEl.type = 'text/css';
  styleEl.appendChild(codeEl);
  document.head.appendChild(styleEl);
}