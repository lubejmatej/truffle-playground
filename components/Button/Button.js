import * as React from "../../_snowpack/pkg/react.js";
import "./Button.css.proxy.js";
const Button = ({
  children,
  type = "button",
  variant = "primary",
  outline = false,
  ...restProps
}) => {
  const classNames = [
    "Button",
    `Button--${variant}`,
    ...outline ? ["Button--outline"] : []
  ];
  return /* @__PURE__ */ React.createElement("button", {
    className: classNames.join(" "),
    type,
    ...restProps
  }, children);
};
export default Button;
