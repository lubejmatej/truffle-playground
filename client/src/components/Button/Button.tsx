import * as React from 'react';

import './Button.css';

const Button: React.FC<
  React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > & {
    variant?: 'primary' | 'secondary';
  }
> = ({ children, variant, ...restProps }) => {
  const btnVariant = variant ?? 'primary';

  return (
    <button
      className={`Button Button--${btnVariant}`}
      type="button"
      {...restProps}
    >
      {children}
    </button>
  );
};

export default Button;
