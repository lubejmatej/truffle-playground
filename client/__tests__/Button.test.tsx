import * as React from 'react';
import {
  create,
  act,
  ReactTestInstance,
  ReactTestRenderer,
} from 'react-test-renderer';

import Button from '../src/components/Button/Button';

const ensureButtonProps = (
  cmpInstance: ReactTestInstance,
  { variant, outline }: { variant?: 'primary' | 'secondary'; outline?: boolean }
) => {
  const buttonProps = cmpInstance.findByType('button').props;
  const buttonClasses = buttonProps.className;

  expect(buttonClasses).toContain('Button');

  if (variant === 'primary') {
    expect(buttonClasses).toContain('Button--primary');
  } else {
    expect(buttonClasses).not.toContain('Button--primary');
  }

  if (variant === 'secondary') {
    expect(buttonClasses).toContain('Button--secondary');
  } else {
    expect(buttonClasses).not.toContain('Button--secondary');
  }

  if (outline) {
    expect(buttonClasses).toContain('Button--outline');
  } else {
    expect(buttonClasses).not.toContain('Button--outline');
  }

  expect(buttonProps.children).toBeUndefined();
};

it('renders button variants', () => {
  let cmpRenderer!: ReactTestRenderer;

  act(() => {
    cmpRenderer = create(<Button />);
  });
  expect(cmpRenderer.toJSON()).toMatchSnapshot();
  ensureButtonProps(cmpRenderer.root, { variant: 'primary', outline: false });

  act(() => {
    cmpRenderer.update(<Button variant="secondary" outline />);
  });
  expect(cmpRenderer.toJSON()).toMatchSnapshot();
  ensureButtonProps(cmpRenderer.root, { variant: 'secondary', outline: true });

  act(() => {
    cmpRenderer.update(<Button outline />);
  });
  expect(cmpRenderer.toJSON()).toMatchSnapshot();
  ensureButtonProps(cmpRenderer.root, { variant: 'primary', outline: true });
});
