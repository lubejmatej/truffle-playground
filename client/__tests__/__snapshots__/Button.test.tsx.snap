// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button variants 1`] = `
<button
  className="Button Button--primary "
  type="button"
/>
`;

exports[`renders button variants 2`] = `
<button
  className="Button Button--secondary Button--outline"
  type="button"
/>
`;

exports[`renders button variants 3`] = `
<button
  className="Button Button--primary Button--outline"
  type="button"
/>
`;
